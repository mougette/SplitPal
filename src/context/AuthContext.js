import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
