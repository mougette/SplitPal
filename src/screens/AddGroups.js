import React, { useState } from 'react';
import {View, Text, StyleSheet, Button, TextInput, FlatList, SafeAreaView, Image, TouchableOpacity} from 'react-native';
import Entry from "../components/EntryComponent";
import SearchAndAdd from "../components/SearchAndAddComponent";
import {Icon} from "react-native-elements";

const DATA = [
  {
    id: '1',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Eliot',
    balance: 'Add',
  },
  {
    id: '2',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Abby',
    balance: 'Add',
  },
  {
    id: '3',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Joe',
    balance: "Add",
  },
  {
    id: '4',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Kevin',
    balance: 'Add',
  },
  {
    id: '5',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Matt',
    balance: 'Add',
  },
  {
    id: '6',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Nick',
    balance: "Add",
  },
  {
    id: '7',
    image: 'https://reactnative.dev/img/tiny_logo.png',
    name: 'Steven',
    balance: 'Add',
  },
];

const AddGroups = ({navigation}) => {

  const [groupName, setGroupName] = useState('');

  const renderItem = ( item ) => {
    return (
        <Entry
            image={item.item.image}
            name={item.item.name}
            balance={item.item.balance}
            onPress = { () => console.log("Yo!") }
        ></Entry>
    );
  };

  return (
      <View style={styles.master}>
        <View style={styles.masterLogo}>
          <Text style={styles.headerBlue}>Create Gro</Text>
          <Text style={styles.headerGray}>ups</Text>
        </View>
        <View style={styles.masterLogo}>
          <Image source ={{uri: 'https://reactnative.dev/img/tiny_logo.png'}}
                 style={{width: 120, height: 120,borderRadius:60}} />
        </View>
        <View style={styles.link}>
          <TextInput
              style={styles.input}
              placeholder= "Group Name"
              onChangeText={setGroupName}
              value={groupName}
              textContentType="emailAddress"
              autoCapitalize="none"
              autoCorrect={false}
          />
          <TouchableOpacity onPress={() => console.log("Group Created")}>
            <Text style={styles.blueTextButton}> Create Group </Text>
          </TouchableOpacity>
        </View>

        <View style={styles.master}>
          <Text style={styles.subHeader}>Add Members</Text>
          <SafeAreaView style={[styles.containerTop, {alignItems: 'center'}]}>
            <FlatList
                data={DATA}
                renderItem={renderItem}
            />
          </SafeAreaView>
        </View>
      </View>

  );
};

const styles = StyleSheet.create({
  master: {
    flex: 1,
    backgroundColor: '#87a1b6'
  },
  masterLogo: {
    padding: 16,
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: '#87a1b6'
  },
  headerBlue: {
    fontSize: 48,
    marginBottom: 10,
    color: '#3B5AAA',
    alignSelf: 'center',
    fontWeight: 'bold',
  },
  headerGray: {
    fontSize: 48,
    marginBottom: 10,
    color: '#454851',
    alignSelf: 'center',
    fontWeight: 'bold',
  },
  header: {
    fontSize: 32,
  },
  subHeader: {
    fontSize: 20,
    paddingLeft: 20,
  },
  blueTextButton: {
    fontSize: 20,
    borderWidth: 1,
    borderRadius: 7,
    overflow: "hidden",
    padding: 10,
    marginLeft: 10,
    color: '#2196F3',
    backgroundColor: 'white',
  },
  link: {
    flexDirection: 'row',
    paddingTop: 20,
  },
  input: {
    width: '50%',
    borderWidth: 1,
    borderRadius: 7,
    padding: 10,
    marginLeft: 20,
    backgroundColor: 'white',
  },
  containerTop: {
    flex: 2,
    marginTop: 15,
  },
  containerBot: {
    flex: 1,
    marginTop: 0,
  },
  item: {
    padding: 20,
    marginVertical: 8,
    paddingHorizontal: '35%',
    borderRadius: 5,
  },
  image: {
    flex: 1,
        justifyContent: "center"
  },
});

export default AddGroups;
