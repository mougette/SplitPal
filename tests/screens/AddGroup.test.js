import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import AddGroups from '../../src/screens/AddGroups'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve("[{\"FirstName\": \"2\", \"LastName\": \"2\", \"Email\": \"abc@e.com\", \"Balance\": \"Split\"}, {\"FirstName\": \"fake\", \"LastName\": \"1\", \"Email\": \"fake1@gmail.com\", \"Balance\": \"Split\"}, {\"FirstName\": \"fake\", \"LastName\": \"5\", \"Email\": \"fake5@gmail.com\", \"Balance\": \"Split\"}, {\"FirstName\": \"hi\", \"LastName\": \"there\", \"Email\": \"newEmail@test.com\", \"Balance\": \"Split\"}, {\"FirstName\": \"Sam\", \"LastName\": \"Hogenson\", \"Email\": \"shogenson@wisc.edu\", \"Balance\": \"Split\"}, {\"FirstName\": \"test\", \"LastName\": \"today\", \"Email\": \"testing@test.com\", \"Balance\": \"Split\"}]"),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Test Hooks', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Render', async () => {
        const {asFragment} = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="AddGroups" component={AddGroups} />
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
       await waitFor(() => {
          expect(asFragment).toMatchSnapshot();
         });
        })
    it('Click Create Group', async () => {
                const {getByText, asFragment} = render(
                <AuthProvider>
                <NavigationContainer>
                <Stack.Navigator>
                <Stack.Screen name="AddGroups" component={AddGroups} />
                </Stack.Navigator>
                </NavigationContainer>
                </AuthProvider>);
                await waitFor(() => {
                fireEvent.press(getByText(" Create Group "));
                expect(asFragment).toMatchSnapshot();
                });
                })
    it('Toggle adding people', async () => {
                    const {getAllByText, asFragment} = render(
                    <AuthProvider>
                    <NavigationContainer>
                    <Stack.Navigator>
                    <Stack.Screen name="AddGroups" component={AddGroups} />
                    </Stack.Navigator>
                    </NavigationContainer>
                    </AuthProvider>);
                    await waitFor(() => {
                    fireEvent.press(getAllByText("Add")[0]);
                    expect(asFragment).toMatchSnapshot();
                    });
                    await waitFor(() => {
                    fireEvent.press(getAllByText("Added")[0]);
                    expect(asFragment).toMatchSnapshot();
                    });
                    })


});