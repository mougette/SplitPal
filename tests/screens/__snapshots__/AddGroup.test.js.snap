// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hooks Click Create Group 1`] = `undefined`;

exports[`Test Hooks Render 1`] = `undefined`;

exports[`Test Hooks Toggle adding people 1`] = `undefined`;

exports[`Test Hooks Toggle adding people 2`] = `undefined`;
